### ISSUE TITLE
SlewRateLimiter parameter strict has no effect

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1798]
**Reported by hansolsson on 9 Oct 2015 15:11 UTC**
The block uses:
```
  if strict then
    der(y) = smooth(1, noEvent(min(max((u - y)/Td, Falling), Rising)));
  else
    der(y) = smooth(1, min(max((u - y)/Td, Falling), Rising));
  end if;
```
But the functions min/max do not generate events (see #601 for rationale); and thus smooth/noEvent is meaningless in the first if-branch.

A simple solution would be:
```
  parameter Boolean strict=true "...";
protected
  Real val=(u-y)/Td;
equation
  if strict then
    der(y)=smooth(1,noEvent(if val<Falling then Falling else if val>Rising then Rising else val));
  else
    der(y)=if val<Falling then Falling else if val>Rising then Rising else val;
  end if;
```
In order to increase compatibility I propose to change the default for 'strict' to 'true'; that will keep the behavior from #529 - similarly as for the other cases in #987 (it was just that the previous behavior was different for this model).

It is possible to avoid the additional variable, but not without code duplication. Using min/max in the strict-branch is possible, but seems unsymmetric.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1798

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 20 Oct 2015 18:07 UTC**
I guess in block VariableLimiter the two cases for the calculation of uMax/uMin can also be omitted.

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 21 Oct 2015 07:44 UTC**
Replying to [comment:1 beutlich]:
> I guess in block VariableLimiter the two cases for the calculation of uMax/uMin can also be omitted.

Yes, I would only use the 
```
 uMax = max(limit1, limit2);
 uMin = min(limit1, limit2);
```
However, the block is confusing - limit1 is described as minimum, but min/max is used to compute actual minimum.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 21 Oct 2015 07:53 UTC**
Yes, I also noticed this. Maybe it's just in case the user mixes limit1 and limit2 up.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 11 Dec 2015 14:51 UTC**
Fixed in 986f8d4ee99af690b5c6861baf9ecf1a225c14c2:
- SlewRateLimiter reformulated with if-clauses instead of min/max (that do not generate events), as suggested
- VariableLimiter: replaced uMax by limiter1 and uMin by limiter2 and added an assert limiter1 > limiter2, as suggested

**Comment by modelica-trac-importer** [comment]
**Changelog modified by otter on 11 Dec 2015 14:51 UTC**
Corrected Nonlinear blocks SlewRateLimiter and VariableLimiter because strict-parameter did not have an effect
